test/models/provider_test.rb

```diff
@@ -0,0 +1,30 @@
+require 'katello_test_helper'
+
+module Katello
+  class ProviderTest < ActiveSupport::TestCase
+    def setup
+      @provider = katello_providers(:redhat)
+      @upstream_params = { 'idCert' => { 'key' => '', 'cert' => ''}}
+      @expected_params = {:upstream_cert => "", :upstream_key => "", :ca_file => nil, :params => {:capabilities => [], :facts => {:distributor_version => "sat-6.3" }}}
+    end
+
+    def test_owner_upstream_update
+      @expected_params[:url] = "https://subscription.rhn.redhat.com/subscription/consumers/"
+      #expected_params = { "url" => "https://subscription.rhn.redhat.com/subscription/consumers/", "upstream_cert" => "", "upstream_key" => "", "ca_file" => nil, "params" => {:capabilities => [], :facts => {:distributor_version => "sat-6.3" }}}
+      Resources::Candlepin::UpstreamConsumer.expects(:update).with(@expected_params[:url], @expected_params[:upstream_cert], @expected_params[:upstream_key], @expected_params[:ca_file], @expected_params[:params])
+
+      @upstream_params['apiUrl'] = 'https://subscription.rhn.redhat.com/subscription/consumers/'
```

Comment:
Instead of duplicating the url what about 
```
@upstream_params['apiUrl'] = @expected_params[:url]
```

Nitpick: how about a shorter url? http://theforeman.org for example